alphadia/workflow/searchoptimization.py

```diff
@@ -58,8 +49,140 @@ def step(self, precursors_df: pd.DataFrame, fragments_df: pd.DataFrame):
         """
         pass
 
+
+class AutomaticOptimizer(BaseOptimizer):
+    def __init__(
+        self,
+        initial_parameter: float,
+        workflow,
+        **kwargs,
+    ):
+        """This class automatically optimizes the search parameter and stores the progres of optimization in a dataframe, history_df.
+
+        Parameters
+        ----------
+        initial_parameter: float
+            The parameter used for search in the first round of optimization.
+
+
+        """
+        super().__init__(workflow, **kwargs)
+        self.history_df = pd.DataFrame()
+        self.workflow.com.fit({self.parameter_name: initial_parameter})
+        self.has_converged = False
+
+    def step(
+        self,
+        precursors_df: pd.DataFrame,
+        fragments_df: pd.DataFrame,
+        current_step: int = -1,
+    ):
+        """See base class. The feature is used to track the progres of the optimization (stored in .feature) and determine whether it has converged."""
+        if self.has_converged:
+            self.reporter.log_string(
+                f"✅ {self.parameter_name:<15}: optimization complete. Optimal parameter {self.workflow.com.__dict__[self.parameter_name]} found after {len(self.history_df)} searches.",
+                verbosity="progress",
+            )
+            return
+
+        new_row = pd.DataFrame(
+            [
+                {
+                    "parameter": float(
```

Comment:
Are the type conversions needed? this is rather uncommon for python and is usually a sign that something is off.

Reply:
No, I have removed them.